tensorflow/java/src/test/java/org/tensorflow/ScopeTest.java

```diff
@@ -0,0 +1,184 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+package org.tensorflow;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link org.tensorflow.Scope}. */
+@RunWith(JUnit4.class)
+public class ScopeTest {
+
+  @Test
+  public void basic() {
+    try (Graph g = new Graph()) {
+      Scope s = Scope.create(g);
+      Const c1 = Const.create(s, 42);
+      assertEquals("Const", c1.input().op().name());
+      Const c2 = Const.create(s, 7);
+      assertEquals("Const_1", c2.input().op().name());
+      Const c3 = Const.create(s.withOpName("four"), 4);
+      assertEquals("four", c3.input().op().name());
```

Comment:
Perhaps also add, for documentation purposes, test for the following scenario (a second constant with the same name):

```java
Const c4 = Const.create(s.withOpName("four"), 4);
```